propsigner/client.go

```diff
@@ -0,0 +1,66 @@
+package propsigner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/textileio/go-auctions-client/gen/wallet"
+)
+
+const (
+	maxResponseMessageSize = 100 << 10
+)
+
+// RequestSignatureV1 request a signature for a deal proposal to a remote wallet.
+func RequestSignatureV1(
+	ctx context.Context,
+	h host.Host,
+	authToken string,
+	proposal market.DealProposal,
+	remoteWallet peer.ID) (*crypto.Signature, error) {
+	proposalCborBytes := &bytes.Buffer{}
+	if err := proposal.MarshalCBOR(proposalCborBytes); err != nil {
+		return nil, fmt.Errorf("marshaling deal proposal to cbor: %s", err)
+	}
+
+	req := pb.ProposalSigningRequest{
+		AuthToken:            authToken,
+		FilecoinDealProtocol: filecoinDealProtocolV1,
+		Payload:              proposalCborBytes.Bytes(),
+	}
+
+	s, err := h.NewStream(ctx, remoteWallet, v1Protocol)
+	if err != nil {
+		return nil, fmt.Errorf("creating libp2p stream: %s", err)
+	}
+	defer func() {
+		if err := s.Close(); err != nil {
+			log.Errorf("closing deal proposal signer stream: %s", err)
+		}
+	}()
+
+	if err := writeMsg(s, &req); err != nil {
```

Comment:
Remember these `writeMsg` and `readMsg` that will be explained in other file.

In short, every time we want to write a serialized proto message, we prepend the bytes with a varint containing the size of the message. This is just to the receiving (or reading) end to know how many bytes should read in the stream to get the serialized message to unmarshal later. (So, `[size_of_msg][msg]`).